text/0000-module-resolution.md

```diff
@@ -0,0 +1,340 @@
+
+```json5
+// package.json
+{
+    "name: "foo",
+    "version": "bar",
+    "lwc": {
+        "modules": [
+            /* list with modules records goes here */
+        ]
+    }
+}
+```
+
+```json5
+// lwc.config.json
+{
+    "modules:" [
+        /* list with modules records goes here */
+    ]
+}
+```
+
+Its important that **any plugin, tool or project honor this configuration**, otherwise the resolution will not work correctly in most environments.
+
+Moreover, note that this type of configuraiton is not something new, many tools use an equivalent configuration (babel, lerna, rollup, webpack, ...)
+
+### Module resolution configuration
+
+The `modules` key on the configuration will hold an Array of `ModuleRecords`. A module record is an object with a specific shape depending on the type of module record it holds. The type will depend on the mode of resolution, we will be defining three types of `ModuleRecords` for now:
+  - AliasModuleRecord: Define an specifier that points to a specific module.
+  - DirectoryModuleRecord: Defines a folder where many namespaced modules can be defined
+  - NpmModuleRecord: Defines a set of packages that will hold more modules to be resolved.
+
+### Why Objects instead of raw strings?
+
+In previous draft versions, `module` entries used to be just *strings* which had slightly better ergonomics. However, raw strings present a lot of ambiguities, let's look at one example:
+
+```json
+{
+    "modules": [
+        "just-string"
+    ]
+}
+```
+
+On the example below, it will be impossible to know if `just-string` is the name of an npm package, or a name of a folder that contains components, hence creationg resolution ambiguity.
+
+We considered other alternatives such as forcing specific prefix like `npm:package-name` or require a final slash (ex. `folder-name/`) for folders, but we believed that those were more cumbersome to learn and future-proof, so standarizing on an consistent object shape as ModuleRecord was a preferable and superior approach.
+
+### Types of ModuleRecord
+
+#### AliasModuleRecord
+
+This type of resolution maps a given module specifier to a particular path within the root of that module.
+
+```json
+{
+    "modules": [
+       {
+            "name": "ui/button",
+            "path": "src/modules/ui/button/button.js"
+        }        
+    ]
+}
+```
+
+`AliasModuleRecord` must contain `name` and a `path` keys.
+
+#### DirectoryModuleRecord
+
+This type of resolution allows to point to to a folder that contain LWC modules.
+Its important to note that within that folder the structure must be very specific, it must contain a namespace folder, then a folder per module bundle (moduleName) adnd them the moduleEntry file that matches the name of the bundle (the entry must be of type .css, .html, .js or .ts)
```

Comment:
> Its important to note that within that folder the structure must be very specific

Also important to note that the compiler does not define the same requirements. I understand why you did not want to impose unnecessary restrictions but maybe we should consider formalizing that.

Reply:
You mentioned above the configuration specificity from inside out. If that part were spec'ed and supported you could allow `glob` pattern matching instead of relying on a particular directory structure. _For example although Jest has it easy because you can match on a file name like `*.test.js`, it still manages to basically load any test module from any matching directory efficiently_

Reply:
I will add some extra clarification here for the first comment: